### PR TITLE
add support for new package cache dir (Pkl 0.26)

### PR DESCRIPTION
**Pull request by stackoverflow**

### CODE REVIEW

**Review by bioball — CHANGES_REQUESTED** [review]
The packages-2 path is slightly different due to encoding, as per https://github.com/apple/pkl-evolution/pull/3.

You can copy [IoUtils.encodePath](https://github.com/apple/pkl/blob/main/pkl-core/src/main/java/org/pkl/core/util/IoUtils.java#L774) out of the core implementation and use that here.

File: src/main/kotlin/org/pkl/intellij/util/Util.kt [hunk 1]
```diff
@@ -45,8 +45,8 @@ val pklDir: VirtualFile?
 val pklCacheDir: VirtualFile?
   get() = pklDir?.findChild("cache")
 
-val packages1CacheDir: VirtualFile?
-  get() = pklCacheDir?.findChild("package-1")
+val packagesCacheDir: VirtualFile?
+  get() = pklCacheDir?.run { findChild("package-2") ?: findChild("package-1") }
```
Comment (holzensp) [inline src/main/kotlin/org/pkl/intellij/util/Util.kt:49]: This commits to the newest cache directory found on the system. For backward compatibility, shouldn't this also depend on the `MinPklVersion` of a module (for example)?

File: src/main/kotlin/org/pkl/intellij/packages/dto/PackageUri.kt [hunk 2]
```diff
@@ -98,9 +98,17 @@ data class PackageUri(
...
+      "package-2/$authority$basePath@$version/$lastSegmentName.zip",
+      "package-1/$authority$basePath@$version/$lastSegmentName.zip"
+    )
 
-  val relativeMetadataFile = "package-1/$authority$basePath@$version/$lastSegmentName.json"
+  val relativeMetadataFiles =
+    listOf(
+      "package-2/$authority$basePath@$version/$lastSegmentName.json",
+      "package-1/$authority$basePath@$version/$lastSegmentName.json"
+    )
```
Comment (holzensp) [inline src/main/kotlin/org/pkl/intellij/packages/dto/PackageUri.kt:111]: How much faith do we have in this repetition being JITed out?
```suggestion
  private val packageFilenameBase = "$authority$basePath@$version/$lastSegmentName"
  
  val relativeZipFiles =
     listOf(
       "package-2/$packageFilenameBase.zip",
       "package-1/$packageFilenameBase.zip"
     )

   val relativeMetadataFiles =
     listOf(
       "package-2/$packageFilenameBase.json",
       "package-1/$packageFilenameBase.json"
     )
```

**Review by holzensp — APPROVED** [review]
No blockers; couple suggestions.

File: src/main/kotlin/org/pkl/intellij/util/Util.kt (hunk 1 above)
Comment (bioball) [reply src/main/kotlin/org/pkl/intellij/util/Util.kt:49]: I think the ability to discover sources is different than providing diagnostics about code correctness. It makes sense to me that searching for sources is a "best-effort" approach, and I don't think this needs to change based on `minPklVersion`.

**Review by bioball — APPROVED** [review]
We paired on this and LGTM!

**Pull request merged**